# Pandas Lab

### Read in the CSV file from a URL

In [2]:
import pandas as pd # we forgot this step!

In [108]:
drinks = pd.read_csv('https://raw.githubusercontent.com/misrab/SG_DAT1/master/data/drinks.csv', na_filter='FALSE')
# we forgot 'na_filter='FALSE'

Are we forgetting something...?

### 1. Show the first 18 rows of drinks

In [130]:
drinks.head(18) # done!

country  beer_servings  spirit_servings  wine_servings  \
0         Afghanistan              0                0              0   
1             Albania             89              132             54   
2             Algeria             25                0             14   
3             Andorra            245              138            312   
4              Angola            217               57             45   
5   Antigua & Barbuda            102              128             45   
6           Argentina            193               25            221   
7             Armenia             21              179             11   
8           Australia            261               72            212   
9             Austria            279               75            191   
10         Azerbaijan             21               46              5   
11            Bahamas            122              176             51   
12            Bahrain             42               63              7   
13         Bangladesh              0                0              0   
14           Barbados            143              173             36   
15            Belarus            142              373             42   
16            Belgium            295               84            212   
17             Belize            263              114              8   

    total_litres_of_pure_alcohol continent  
0                            0.0        AS  
1                            4.9        EU  
2                            0.7        AF  
3                           12.4        EU  
4                            5.9        AF  
5                            4.9        NA  
6                            8.3        SA  
7                            3.8        EU  
8                           10.4        OC  
9                            9.7        EU  
10                           1.3        EU  
11                           6.3        NA  
12                           2.0        AS  
13                           0.0        AS  
14                           6.3        NA  
15                          14.4        EU  
16                          10.5        EU  
17                           6.8        NA

### 2. Create a variable called beer_servings and use it to store the beer_servings column

In [110]:
beer_servings = drinks.beer_servings # assigning the variable
type(beer_servings)

pandas.core.series.Series

In [111]:
beer_servings # done!

0        0
1       89
2       25
3      245
4      217
5      102
6      193
7       21
8      261
9      279
10      21
11     122
12      42
13       0
14     143
15     142
16     295
17     263
18      34
19      23
20     167
21      76
22     173
23     245
24      31
25     231
26      25
27      88
28      37
29     144
      ... 
163    128
164     90
165    152
166    185
167      5
168      2
169     99
170    106
171      1
172     36
173     36
174    197
175     51
176     51
177     19
178      6
179     45
180    206
181     16
182    219
183     36
184    249
185    115
186     25
187     21
188    333
189    111
190      6
191     32
192     64
Name: beer_servings, dtype: int64

### 3. Display a dataframe where the only rows are those with continent North America

In [127]:
drinks[drinks.continent=='NA'] # done!

country  beer_servings  spirit_servings  \
5               Antigua & Barbuda            102              128   
11                        Bahamas            122              176   
14                       Barbados            143              173   
17                         Belize            263              114   
32                         Canada            240              122   
41                     Costa Rica            149               87   
43                           Cuba             93              137   
50                       Dominica             52              286   
51             Dominican Republic            193              147   
54                    El Salvador             52               69   
68                        Grenada            199              438   
69                      Guatemala             53               69   
73                          Haiti              1              326   
74                       Honduras             69               98   
84                        Jamaica             82               97   
109                        Mexico            238               68   
122                     Nicaragua             78              118   
130                        Panama            285              104   
143             St. Kitts & Nevis            194              205   
144                     St. Lucia            171              315   
145  St. Vincent & the Grenadines            120              221   
174             Trinidad & Tobago            197              156   
184                           USA            249              158   

     wine_servings  total_litres_of_pure_alcohol continent  
5               45                           4.9        NA  
11              51                           6.3        NA  
14              36                           6.3        NA  
17               8                           6.8        NA  
32             100                           8.2        NA  
41              11                           4.4        NA  
43               5                           4.2        NA  
50              26                           6.6        NA  
51               9                           6.2        NA  
54               2                           2.2        NA  
68              28                          11.9        NA  
69               2                           2.2        NA  
73               1                           5.9        NA  
74               2                           3.0        NA  
84               9                           3.4        NA  
109              5                           5.5        NA  
122              1                           3.5        NA  
130             18                           7.2        NA  
143             32                           7.7        NA  
144             71                          10.1        NA  
145             11                           6.3        NA  
174              7                           6.4        NA  
184             84                           8.7        NA

### 4. Create a new dataframe called north_america that holds your answer in 1.

*drinks (the original dataframe) should remain unchanged*

In [114]:
north_america=drinks[drinks.continent=='NA']

In [115]:
type(north_america)

pandas.core.frame.DataFrame

In [116]:
north_america.head(18) # done!

country  beer_servings  spirit_servings  wine_servings  \
5     Antigua & Barbuda            102              128             45   
11              Bahamas            122              176             51   
14             Barbados            143              173             36   
17               Belize            263              114              8   
32               Canada            240              122            100   
41           Costa Rica            149               87             11   
43                 Cuba             93              137              5   
50             Dominica             52              286             26   
51   Dominican Republic            193              147              9   
54          El Salvador             52               69              2   
68              Grenada            199              438             28   
69            Guatemala             53               69              2   
73                Haiti              1              326              1   
74             Honduras             69               98              2   
84              Jamaica             82               97              9   
109              Mexico            238               68              5   
122           Nicaragua             78              118              1   
130              Panama            285              104             18   

     total_litres_of_pure_alcohol continent  
5                             4.9        NA  
11                            6.3        NA  
14                            6.3        NA  
17                            6.8        NA  
32                            8.2        NA  
41                            4.4        NA  
43                            4.2        NA  
50                            6.6        NA  
51                            6.2        NA  
54                            2.2        NA  
68                           11.9        NA  
69                            2.2        NA  
73                            5.9        NA  
74                            3.0        NA  
84                            3.4        NA  
109                           5.5        NA  
122                           3.5        NA  
130                           7.2        NA

### 5.  What is the average wine consumption per person per year in Africa?

In [51]:
drinks.wine_servings[drinks.continent=='AF'].mean() # done!

16.264150943396228

Average Wine consumption per person per year in Africa = 16.26

### 6.  Show a list of the top 10 spirit drinking countries.

In [136]:
drinks[['country', 'spirit_servings']].sort_values(by='spirit_servings').tail(11) # done!

country  spirit_servings
25             Bulgaria              252
40         Cook Islands              254
169            Thailand              258
50             Dominica              286
155            Slovakia              293
72               Guyana              302
144           St. Lucia              315
73                Haiti              326
141  Russian Federation              326
15              Belarus              373
68              Grenada              438

### 7.  What is the average wine consumption in South America?

In [14]:
drinks.wine_servings[drinks.continent=='SA'].mean() # done!

62.416666666666664

Average Wine consumption in SA = 62.416

In [117]:
drinks.groupby('continent').mean() # double check

beer_servings  spirit_servings  wine_servings  \
continent                                                  
AF             61.471698        16.339623      16.264151   
AS             37.045455        60.840909       9.068182   
EU            193.777778       132.555556     142.222222   
NA            145.434783       165.739130      24.521739   
OC             89.687500        58.437500      35.625000   
SA            175.083333       114.750000      62.416667   

           total_litres_of_pure_alcohol  
continent                                
AF                             3.007547  
AS                             2.170455  
EU                             8.617778  
NA                             5.995652  
OC                             3.381250  
SA                             6.308333

### 8.  Which continent has the highest on average wine consumption?

In [126]:
drinks.groupby('continent').wine_servings.mean().sort_values(ascending=False).head(1) # done!

continent
EU    142.222222
Name: wine_servings, dtype: float64

EU has the highest average wine consumption with 142.22